### PR TITLE
Fixed #3386

### LINKED ISSUE #3386: Wither effect deaths crash server
Client version:   1.10.2
Server OS:        Linux
Commit id:        b268db4caae1b5612ae0f3adabf164efc06de368
### Expected behavior

Player dies normally from wither effect, without causing a server crash.
### Actual behavior

When a player dies while having the wither effect, the server crashes.
### Steps to reproduce the behavior

Apply the wither effect with
`Player:AddEntityEffect(20, 9999, 1, 1)`
in a plugin, and wait for your player to die.

### PR DESCRIPTION
**Pull request by stone3311**
Fixed issue #3386.

`Effect->OnTick(*this);` caused the iterator to become invalid. This issue is fixed by saving the effect and calling `OnTick` after the loop.

### CODE REVIEW

**Comment by sphinxc0re** [comment]
Does this only fix the one issue? Are there more issues leading back to the same problem?

File: src/Entities/Pawn.cpp [hunk 1]
```diff
@@ -48,14 +48,17 @@ void cPawn::Destroyed()
 
 void cPawn::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 {
+	std::vector<cEntityEffect*> EffectsToTick;
```
Comment (madmaxoft) [inline src/Entities/Pawn.cpp]: Style: add space before the `*`

File: src/Entities/Pawn.cpp [hunk 2]
```diff
@@ -69,6 +72,12 @@ void cPawn::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 		// TODO: Check for discrepancies between client and server effect values
 	}
 
+
+	for (cEntityEffect * Effect : EffectsToTick)
```
Comment (madmaxoft) [inline src/Entities/Pawn.cpp]: Use `auto` or even `const auto`, if possible.

**Comment by madmaxoft** [comment]
This is definitely a needed change and there might be more issues caused by that error, not necessarily reported.

**Comment by sphinxc0re** [comment]
Why is CircleCI dumb again? :grin:

**Comment by madmaxoft** [comment]
I've just checked it and found out they had added an option not to build PRs from forked repos and made it the default, I reset it, so now the builds should work again. We'll see when @stone3311 pushes the changes I requested to his branch, it should trigger the rebuild.

**Comment by stone3311** [comment]
The crash had only happened when dying due to the wither effect. Getting killed or issuing /kill in the console while having the wither effect did not crash the server.

**Comment by sphinxc0re** [comment]
you should rebase your branch onto the current master @stone3311

**Comment by stone3311** [comment]
Ok, I rebased it onto the current master @sphinxc0re

**Comment by madmaxoft** [comment]
Nah, there was no need to rebase, this is a change to be squash-merged anyway. And you didn't rebase onto master, you merged master (which we don't like doing, it creates loops in the history). Thanks for trying, anyway. Merging now.

**Comment by stone3311** [comment]
Sorry about that.

**Pull request merged**